services/oldsService.js

```diff
@@ -15,6 +15,104 @@ const inArray = (arr, obj) => {
     return false
 }
 
+const sqlTime = (workerId, start, end) => `
+    SELECT
+        IFNULL(x.jobID, -100) as jobID,
+        secs,
+        hours,
+        time
+    FROM (
+    SELECT
+        t.jobID,
+        SUM(UNIX_TIMESTAMP(t.endDTS) - UNIX_TIMESTAMP(t.startDTS)) as secs,
+        ROUND(SUM(UNIX_TIMESTAMP(t.endDTS) - UNIX_TIMESTAMP(t.startDTS))/3600,2) as hours,
+            CONCAT(
+                LPAD(SUM(UNIX_TIMESTAMP(t.endDTS) - UNIX_TIMESTAMP(t.startDTS)) DIV 3600 % 60,2, 0),
+                ':',
+                LPAD(SUM(UNIX_TIMESTAMP(t.endDTS) - UNIX_TIMESTAMP(t.startDTS)) % 3600 DIV 60, 2, 0),
+                ':00'
+                ) as time
+    FROM TIMINGS t
+        JOIN JOBS j
+        ON t.jobID=j.id
+    WHERE t.workerID=${workerId} AND t.startDTS>='${start}' AND t.startDTS<'${end}'
+    GROUP BY t.jobID -- WITH ROLLUP
+    ) as x LEFT JOIN oldsdb.JOBS j on j.id=x.jobID;
```

Comment:
Do not understand this whole query.
In sub-select you do  INNER join jobs=timings by jobId, which means there should be NO nulls in jobId afterwards.
Why do you do left join with jobs again?
Why do you do select from select?